model/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/idp/SamlServiceProvider.java

```diff
@@ -149,58 +149,71 @@ public boolean configIsValid() {
-        result = prime * result + ((entityId == null) ? 0 : entityId.hashCode());
+        result = prime * result + (config == null ? 0 : config.hashCode());
+        result = prime * result + (created == null ? 0 : created.hashCode());
+        result = prime * result + (id == null ? 0 : id.hashCode());
+        result = prime * result + (lastModified == null ? 0 : lastModified.hashCode());
+        result = prime * result + (name == null ? 0 : name.hashCode());
+        result = prime * result + (entityId == null ? 0 : entityId.hashCode());
         result = prime * result + version;
         return result;
     }
 
     @Override
     public boolean equals(Object obj) {
-        if (this == obj)
+        if (this == obj) {
             return true;
-        if (obj == null)
+        }
+        if (obj == null) {
             return false;
-        if (getClass() != obj.getClass())
+        }
+        if (getClass() != obj.getClass()) {
             return false;
+        }
         SamlServiceProvider other = (SamlServiceProvider) obj;
         if (config == null) {
-            if (other.config != null)
+            if (other.config != null) {
                 return false;
-        } else if (!config.equals(other.config))
+            }
+        } else if (!config.equals(other.config)) {
             return false;
+        }
         if (created == null) {
-            if (other.created != null)
+            if (other.created != null) {
                 return false;
-        } else if (!created.equals(other.created))
+            }
+        } else if (!created.equals(other.created)) {
             return false;
+        }
         if (id == null) {
-            if (other.id != null)
+            if (other.id != null) {
                 return false;
-        } else if (!id.equals(other.id))
+            }
+        } else if (!id.equals(other.id)) {
             return false;
+        }
         if (lastModified == null) {
-            if (other.lastModified != null)
+            if (other.lastModified != null) {
                 return false;
-        } else if (!lastModified.equals(other.lastModified))
+            }
+        } else if (!lastModified.equals(other.lastModified)) {
             return false;
+        }
         if (name == null) {
-            if (other.name != null)
+            if (other.name != null) {
                 return false;
-        } else if (!name.equals(other.name))
+            }
+        } else if (!name.equals(other.name)) {
             return false;
+        }
         if (entityId == null) {
-            if (other.entityId != null)
+            if (other.entityId != null) {
                 return false;
-        } else if (!entityId.equals(other.entityId))
-            return false;
-        if (version != other.version)
+            }
+        } else if (!entityId.equals(other.entityId)) {
             return false;
-        return true;
+        }
+        return !(version != other.version);
```

Comment:
See here logic from your tool or was there before, but can be simplified

Reply:
previous was
``` 
  if (version != other.version)
            return false;
        return true;
``` 

I rebased develop, and added a few more fixes, now:
```
        return version == other.version;
```